homeex/src/main/java/io/codelex/streams/practice/StreamsExercise.java

```diff
@@ -0,0 +1,190 @@
+package io.codelex.streams.practice;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * TODO - search for StreamsExerciseTest and make all tests pass
+ */
+public class StreamsExercise {
+    public static List<Integer> returnSquareRoot(List<Integer> numbers) {
+        return numbers.stream()
+                .map(Math::sqrt)
+                .map(Double::intValue)
+                .collect(toList());
+    }
+
+    public static List<Integer> getAgeFromUsers(List<User> user) {
+        return user.stream()
+                .map(User::getAge)
+                .collect(toList());
+    }
+
+    public static List<Integer> getDistinctAges(List<User> users) {
+        return users.stream()
+                .map(User::getAge)
+                .distinct()
+                .collect(toList());
+    }
+
+    public static List<User> getLimitedUserList(List<User> users, int limit) {
+        return users.stream()
+                .limit(limit)
+                .collect(toList());
+    }
+
+    public static Integer countUsersOlderThen25(List<User> users) {
+        List<User> result = users.stream()
+                .filter(q -> q.getAge() > 25)
+                .toList();
```

Comment:
Šī vietā var pasaukt .count() un tad uzreiz būs rezultāts, nevajadzēs to .size vēlāk

Reply:
Tad metode atgriež Long.

Izlaboju visas metodes.
Pārkopēju šo uzdevumu un cita projekta jo te kaut kas nestrādā ar Maven, redzēs kā būs pēc iemergošanas